drools/src/main/java/com/datatorrent/drools/rules/DroolsKnowledgeJarLoader.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017 DataTorrent, Inc. ALL Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.drools.rules;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+public class DroolsKnowledgeJarLoader
+{
+  public static void addkjarToClasspath(String kjarPath) throws IOException
+  {
+    try {
+      File localRulesDir = copykJartoLocalFS(kjarPath);
+      String[] kjarFiles = localRulesDir.list();
+      URL[] urls = new URL[kjarFiles.length];
+      int i = 0;
+      for (String kjarFilePath : kjarFiles) {
+        urls[i++] = new File(localRulesDir, kjarFilePath).toURI().toURL();
+      }
+      ClassLoader loader = Thread.currentThread().getContextClassLoader();
+      URLClassLoader urlClassLoader = URLClassLoader.newInstance(urls, loader);
+      Thread.currentThread().setContextClassLoader(urlClassLoader);
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private static File copykJartoLocalFS(String kjarPath) throws IOException
+  {
+    FileSystem fs = getFSInstance(new Path(kjarPath));
+    File localRulesDir = createTempLocalRulesDir();
+
+    FileStatus rules = fs.getFileStatus(new Path(kjarPath));
+    if (rules.isDirectory()) {
+      for (FileStatus ruleFile : fs.listStatus(new Path(kjarPath))) {
+        if (ruleFile.isFile()) {
+          File localRulesFile = new File(localRulesDir, ruleFile.getPath().getName());
+          fs.copyToLocalFile(false, ruleFile.getPath(), new Path(localRulesFile.getAbsolutePath()), true);
+        }
+      }
+    } else {
+      File localRulesFile = new File(localRulesDir, rules.getPath().getName());
+      fs.copyToLocalFile(false, rules.getPath(), new Path(localRulesFile.getAbsolutePath()), true);
+    }
+
+    return localRulesDir;
+  }
+
+  private static File createTempLocalRulesDir() throws IOException
+  {
+    File localRulesDir = File.createTempFile("drools-rules-", "");
```

Comment:
This will create a **file** in app launch temp directory. But as we need a dir, in next steps I delete this file and create a directory with this name.
I didn't find a way to directly create a temp directory, this was suggested way in different forums.